### ISSUE TITLE
Add permalink, including view preferences

### ISSUE THREAD

**Issue by tripu** [w3c/mailing-list-archives#27]
Instead of #24, have a &ldquo;permalink&rdquo; or &ldquo;link to this page&rdquo; somewhere on the page.

That URI will include the current &ldquo;view settings&rdquo; (thread view, formatting, extended headers, etc), either as fragments (`⋯/0002.html#thread=on`) or as parameters (`⋯/0002.html?thread=on&headers=simple&formatting=md`). @gosko seems to prefer the former, for cacheability&nbsp;&mdash;&nbsp;but we have to think how to encode more than one tuple as a _fragment_, and also decide which is best for crawlers, non-JS UAs, etc.

**Comment by gosko** [comment]
I don't think we need to include parameters for things like the short/detailed header toggle or nice/plain formatting, maybe just `0002.html` for a single message, `0002.html?view=thread` for the entire thread, and `0002.html?view=flattened` for the flattened view? (and maybe we can come up with a better name than 'flattened': inline? chrono?)

**Comment by tripu** [comment]
> I don't think we need to include parameters for things like the short/detailed header toggle or nice/plain formatting,

Maybe not. But I'd like to have a mechanism that can be easily extended. It's possible that we will want to encode more than one piece of configuration in the URL at some point; if we are going to include display options in the URL, we better do it in a way that is not limited.

> Maybe we can come up with a better name than 'flattened': inline? chrono?

I suspect the name will present itself when that feature is implemented and we know exactly how it works :)

**Comment by gosko** [comment]
also, as you might have guessed from the inconsistencies between my previous comment and my comments in #24 I am not completely sure whether to use fragments or query strings for things like this; maybe an implementation will reveal the pros/cons of each :)
